airflow/auth/managers/base_auth_manager.py

```diff
@@ -63,6 +68,69 @@ def get_user_id(self) -> str:
     def is_logged_in(self) -> bool:
         """Return whether the user is logged in."""
 
+    @abstractmethod
+    def is_authorized(
+        self,
+        action: ResourceAction,
+        resource_type: str,
```

Comment:
Should we make it enum rather that str ? That seems like a closed list of resource types we can have permissions for?

Reply:
Haha, it has already been called out by @vandonr-amz, I was thinking of doing it in a separate PR but 2 callouts seems enough to justify it doing it here. I'll do it here